src/IConnection.php

```diff
@@ -38,6 +38,11 @@ public function query(...$args);
 	 */
 	public function queryArgs($query, array $args = []);
 
+	/**
+	 * @param QueryBuilder $queryBuilder
+	 * @return Result
+	 */
+	public function queryFrom(QueryBuilder $queryBuilder): Result;
```

Comment:
Please:
- rename the method to queryByQueryBuilder(). 
- strip the phpdoc, which is useless
- add an empty line between method to match the coding style (2 empty lines)
- fix the tests, you have wrong namespace in Conneciton class :)

Otherwise 👍 , this is something what I wanted do before 3.0 :)